relayer/message_relayer.go

```diff
@@ -95,45 +92,45 @@ func (r *messageRelayer) relayMessage(warpMessageInfo *vmtypes.WarpMessageInfo,
 
 	startCreateSignedMessageTime := time.Now()
 	// Query nodes on the origin chain for signatures, and construct the signed warp message.
-	signedMessage, err := r.createSignedMessage(requestID)
+	signedMessage, err := r.createSignedMessage(requestID, destinationChainID)
 	if err != nil {
 		r.logger.Error(
 			"Failed to create signed warp message",
 			zap.Error(err),
 		)
-		r.incFailedRelayMessageCount("failed to create signed warp message")
+		r.incFailedRelayMessageCount("failed to create signed warp message", destinationChainID)
 		return err
 	}
 
 	// create signed message latency (ms)
-	r.setCreateSignedMessageLatencyMS(float64(time.Since(startCreateSignedMessageTime).Milliseconds()))
+	r.setCreateSignedMessageLatencyMS(float64(time.Since(startCreateSignedMessageTime).Milliseconds()), destinationChainID)
 
-	err = messageManager.SendMessage(signedMessage, warpMessageInfo.WarpPayload, r.destinationChainID)
+	err = messageManager.SendMessage(signedMessage, warpMessageInfo.WarpPayload, destinationChainID)
 	if err != nil {
 		r.logger.Error(
 			"Failed to send warp message",
 			zap.Error(err),
 		)
-		r.incFailedRelayMessageCount("failed to send warp message")
+		r.incFailedRelayMessageCount("failed to send warp message", destinationChainID)
 		return err
 	}
 	r.logger.Info(
 		"Finished relaying message to destination chain",
-		zap.String("destinationChainID", r.destinationChainID.String()),
+		zap.String("destinationChainID", destinationChainID.String()),
 	)
-	r.incSuccessfulRelayMessageCount()
+	r.incSuccessfulRelayMessageCount(destinationChainID)
 	return nil
 }
 
 // Run collects signatures from nodes by directly querying them via AppRequest, then aggregates the signatures, and constructs the signed warp message.
-func (r *messageRelayer) createSignedMessage(requestID uint32) (*warp.Message, error) {
+func (r *messageRelayer) createSignedMessage(requestID uint32, destinationChainID ids.ID) (*warp.Message, error) {
 	r.logger.Info(
 		"Starting relayer routine",
-		zap.String("destinationChainID", r.destinationChainID.String()),
+		zap.String("destinationChainID", destinationChainID.String()),
 	)
 
 	// Get the current canonical validator set of the source subnet.
-	validatorSet, totalValidatorWeight, err := r.getCurrentCanonicalValidatorSet()
+	validatorSet, totalValidatorWeight, err := r.getCurrentCanonicalValidatorSet(destinationChainID)
```

Comment:
Digging in to the implementation of `getCurrentCanonicalValidatorSet`, I see why it's necessary to now pass the `destinationChainID`, but looking at the call here, I was confused why we pass the destination chain ID when  trying to look up the source subnet validator set.

Could we make this more intuitive by creating a helper along the lines of "determineSigningSubnetD" that performs the check on whether or not the message was sent from the primary network, and then have `getCurrentCanonicalValidatorSet` take that subnet ID as a parameter to look up the validator set for?